onnxscript/tensor.py

```diff
@@ -73,40 +77,82 @@ def __getitem__(self, index):
         op = self._opset
         if op.version < 13:
             raise RuntimeError("Indexing requires opset 13 or later.")
-        if isinstance(index, int):
-            # case A[i]: indexing
-            # promote integer input to tensor
-            i = Tensor(np.array(index))
-            # use Gather to perform indexing
-            return op.Gather(self, i, axis=0)
-        if not isinstance(index, (slice, tuple)):
-            raise TypeError(f"Unexpected type {type(index)} for index.")
-        # case A[i:j] or A[i:j:k], A[i:k, j:l]
-        if isinstance(index, slice):
-            # Treat 1-dimensional slicing A[i:j] as generic n-dimensional case A[i:j,...]
+        if not isinstance(index, tuple):
+            # Normalize representation to a tuple.
+            # A single index-value is equivalent to a tuple with a single element.
             index = (index,)
+        if len(index) > self.rank:
+            raise ValueError(
+                f"Number of indices {len(index)} is greater than rank {self.rank}"
+            )
+        from onnxscript import autocast
```

Comment:
Was this the circular import issue?

Reply:
I put it here to best handle the circularity. (My question the other day was about another issue relating to circular dependence.)

Reply:
In that case I suggest putting it on top unless there is an absolute need. Hiding imports in a function delays any import errors that should happen at load time and makes refactoring harder.

Reply:
Putting it on top will create a warning about the circular import. How does that help? It clutters up the messages, and becomes a false-positive. Putting it here, in fact, clarifies that it is a false-positive. False-positives also have the effect of making users simply ignore such messages.

What refactoring is impacted by this?

Reply:
Following up our offline discussion: the main reason I have it here is to avoid "noise" due to warnings about cyclic-imports. If we can turn that noise off, that would address what bothers me. (This is based on my impression from the github repo settings a while back, not sure if it is still the same.)